### ISSUE TITLE
Weird figure reconstruction results for newly trained model

### ISSUE THREAD

**Issue by 5agado** [podgorskiy/ALAE#47]
I trained a new model on a personal footwear dataset. Sample results from training looked good, but when I run  a `make_figures` script, I obtained this weird, oversaturated/false-colors results. 

![reconstructions_0](https://user-images.githubusercontent.com/1591484/85045173-86970680-b186-11ea-96c2-048e51c7e829.png)

Any idea what's happening?

**Comment by shahik** [comment]
did you train it from scratch or using transfer learning? what was your dataset size?

**Comment by 5agado** [comment]
Trained from scratch, ~100k images.
The weird thing is that the samples during training are good

![sample_129_0](https://user-images.githubusercontent.com/1591484/86040257-a0b0ce80-ba3b-11ea-8e46-c452d013a4ff.jpg)

**Comment by MagicalForcee** [comment]
what are the steps to train a model for custom dataset?

**Comment by uhiu** [comment]
hi, I also encountered this problem. Had you solved it?
![image](https://user-images.githubusercontent.com/29622719/101238510-bbd6e700-371b-11eb-8363-4cdbd588fccb.png)

**Comment by uhiu** [comment]
I think I find the reason. For me, it's because I stop the training at `LOD=5`, where the final `LOD` should be 6. So I should adjust the code in the demo python file like  
```python
# Z, _ = model.encode(x, layer_count - 1, 1)
Z, _ = model.encode(x, layer_count - 2, 1)
# cause the layer_cout=7 in the config file, and I want it to be 5, not 6
``` 
Accordingly, we should also adjust the decoder part
```python
model.decoder(x, 5, 1, noise=True)
```
hope it helps. stay safe

**Comment by podgorskiy** [comment]
Yes, what @uhiu says seems to be the most likely cause.
When training on custom data, make sure that the final LOD is consistent everywhere.

The first thing to check is the config. There are two parameters:

for example from bedroom:


`DATASET.MAX_RESOLUTION_LEVEL: 8`    this means that it will train up to 2**8 resolution (256)
`MODEL.LAYER_COUNT: 7` this means that the network will have 7 blocks. We start from 4x4 and each block doubles the resolution except for the first one. This means the final output will be    4 * 2 ** (7 - 1), which is 256.


Basically, if you want resolution 2**x, then you should set `DATASET.MAX_RESOLUTION_LEVEL: x` and `MODEL.LAYER_COUNT: x-1`

@5agado,
Seems that the very last layer has weights with random initialization.
